content/apps/create-app/callback-nodejs.md

```diff
@@ -1,53 +1,59 @@
 ```js [callback:NodeJS]
 
-params // data from your request handling
 storage // your own memory system
+requestData // data from your request handling
 
-// Retrieve GET query params from your own framework / http handler
-const { code, state } = params;
+const akeneoClientSecret = 'AKENEO_CLIENT_SECRET';
+const akeneoClientId = 'AKENEO_CLIENT_ID';
 
-// Retrieve your app's Client ID with your own logic
-const pimUrl = storage.get("PIM_URL");
-const appState = storage.get("APP_STATE");
-const clientId = storage.get("CLIENT_ID");
-const clientSecret = storage.get("CLIENT_SECRET");
+const pimUrl = storage.get('PIM_URL');
+const state = requestData.state;
+const authorizationCode = requestData.code;
 
 // Control the security state integrity previously defined, to avoid attacks
-if (state !== appState) {
-    return response(403, 
-        {
-            "error": "Forbidden",
-            "error_description": "State integrity failed",
-        }
-    )
+if (!state || state != storage.get('APP_STATE')) {
+    throw new LogicError('Invalid state');
+}
+
+if (!authorizationCode) {
+    throw new LogicError('Missing authorization code');
```

Comment:
Don't use `LogicError`: it is a non-defined custom class. Keep response, it is better.

Reply:
switched to Error since we now have error handling from express controllers